spec/spec_helper_acceptance.rb

```diff
@@ -22,14 +22,28 @@
   c.before :suite do
     # Install module and dependencies
     hosts.each do |host|
-      if fact_on(host, 'osfamily') == 'RedHat'
+      if fact_on(host, 'os.family') == 'RedHat'
         # don't delete downloaded rpm for use with BEAKER_provision=no +
         # BEAKER_destroy=no
         on host, 'sed -i "s/keepcache=.*/keepcache=1/" /etc/yum.conf'
         # refresh check if cache needs refresh on next yum command
         on host, 'yum clean expire-cache'
         # We always need EPEL
         on host, puppet('resource', 'package', 'epel-release', 'ensure=installed')
+
+        major = fact_on(host, 'os.release.major')
+
+        if major == '8'
+          on host, 'dnf -y module enable pki-core'
```

Comment:
What's in the pki-core module? Does that mean we need that module enabled on every candlepin install?

Reply:
```shell
[root@centos8-64-1 ~]# dnf module info pki-core
Last metadata expiration check: 0:04:02 ago on Mon 23 Mar 2020 05:04:27 PM UTC.
Name        : pki-core
Stream      : 10.6 [e] [a]
Version     : 8010020191120041437
Context     : 8ba0ffbe
Repo        : AppStream
Summary     : PKI Core module for PKI 10.6 or later
Description : A module for PKI Core packages for PKI version 10.6 or later.
Artifacts   : jss-0:4.6.0-5.module_el8.1.0+238+005506d1.src
            : jss-0:4.6.0-5.module_el8.1.0+238+005506d1.x86_64
            : jss-debuginfo-0:4.6.0-5.module_el8.1.0+238+005506d1.x86_64
            : jss-debugsource-0:4.6.0-5.module_el8.1.0+238+005506d1.x86_64
            : jss-javadoc-0:4.6.0-5.module_el8.1.0+238+005506d1.x86_64
            : ldapjdk-0:4.21.0-1.module_el8.1.0+238+005506d1.noarch
            : ldapjdk-0:4.21.0-1.module_el8.1.0+238+005506d1.src
            : ldapjdk-javadoc-0:4.21.0-1.module_el8.1.0+238+005506d1.noarch
            : pki-base-0:10.7.3-1.module_el8.1.0+238+005506d1.noarch
            : pki-base-java-0:10.7.3-1.module_el8.1.0+238+005506d1.noarch
            : pki-ca-0:10.7.3-1.module_el8.1.0+238+005506d1.noarch
            : pki-core-0:10.7.3-1.module_el8.1.0+238+005506d1.src
            : pki-kra-0:10.7.3-1.module_el8.1.0+238+005506d1.noarch
            : pki-server-0:10.7.3-1.module_el8.1.0+238+005506d1.noarch
            : pki-symkey-0:10.7.3-1.module_el8.1.0+238+005506d1.x86_64
            : pki-symkey-debuginfo-0:10.7.3-1.module_el8.1.0+238+005506d1.x86_64
            : pki-tools-0:10.7.3-1.module_el8.1.0+238+005506d1.x86_64
            : pki-tools-debuginfo-0:10.7.3-1.module_el8.1.0+238+005506d1.x86_64
            : python3-pki-0:10.7.3-1.module_el8.1.0+238+005506d1.noarch
            : tomcatjss-0:7.4.1-1.module_el8.1.0+238+005506d1.noarch
            : tomcatjss-0:7.4.1-1.module_el8.1.0+238+005506d1.src

Hint: [d]efault, [e]nabled, [x]disabled, [i]nstalled, [a]ctive]
```

The important part above is the `tomcatjss` -- without it, we get:

```
An error occurred in a `before(:suite)` hook.
Failure/Error: on host, 'yum -y install candlepin-selinux'
Beaker::Host::CommandFailure:
  Host 'centos8-64-1' exited with 1 running:
   yum -y install candlepin-selinux
  Last 10 lines of output were:
  	Repository 'candlepin' is missing name in configuration, using id.
  	candlepin                                        12 kB/s |  13 kB     00:01    
  	Extra Packages for Enterprise Linux Modular 8 -  50 kB/s | 117 kB     00:02    
  	Extra Packages for Enterprise Linux 8 - x86_64  1.8 MB/s | 6.1 MB     00:03    
  	(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
  	Error: 
  	 Problem: package candlepin-selinux-3.1.3-1.el8.noarch requires candlepin = 3.1.3-1.el8, but none of the providers can be installed
  	  - package candlepin-3.1.3-1.el8.noarch requires tomcatjss >= 7.2.1-7.1, but none of the providers can be installed
  	  - conflicting requests
  	  - package tomcatjss-7.4.1-1.module_el8.1.0+238+005506d1.noarch is excluded
```

Thinking about it, I am not sure if the `pki-core` module can be installed as an RPM dependency... do you think it makes better sense to tackle it here rather than in packaging, @ekohl ?

Reply:
I'm not familiar enough with modularity to say, but this should at least be on the list of things to be aware of.